console/v1/types_console_plugin.go

```diff
@@ -0,0 +1,197 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +openshift:compatibility-gen:level=1
+
+// ConsolePlugin is an extension for customizing OpenShift web console by
+// dynamically loading code from another service running on the cluster.
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+type ConsolePlugin struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// +kubebuilder:validation:Required
+	Spec ConsolePluginSpec `json:"spec"`
+}
+
+// ConsolePluginSpec is the desired plugin configuration.
+type ConsolePluginSpec struct {
+	// displayName is the display name of the plugin.
+	// The dispalyName should be between 1 and 128 characters.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=128
+	DisplayName string `json:"displayName"`
+	// service is a Kubernetes Service that exposes the plugin using a
+	// deployment with an HTTP server. The Service must use HTTPS and
+	// Service serving certificate. The console backend will proxy the
+	// plugins assets from the Service using the service CA bundle.
+	// +kubebuilder:validation:Required
+	Service ConsolePluginService `json:"service"`
+	// proxy is a list of proxies that describe various service type
+	// to which the plugin needs to connect to.
+	// +kubebuilder:validation:Optional
+	Proxy []ConsolePluginProxy `json:"proxy,omitempty"`
+	// i18n is the configuration of plugin's localization resources.
+	// +kubebuilder:validation:Required
+	I18n ConsolePluginI18n `json:"i18n"`
+}
+
+// LoadType is an enumeration of i18n loading types
+// +kubebuilder:validation:Pattern=`^(Preload|Lazy)$`
```

Comment:
You can use an enum validation for this
```suggestion
// +kubebuilder:validation:Enum:=Preload;Lazy
```